test/test_cuda.jl

```diff
@@ -11,16 +11,41 @@ isdir(outdir) && rm(outdir, recursive = true)
 
 EXAMPLES_DIR = joinpath(examples_dir(), "p4est_2d_dgsem")
 
-@trixi_testset "elixir_advection_basic_gpu.jl" begin
+@trixi_testset "elixir_advection_basic_gpu.jl native" begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_basic_gpu.jl"),
+                        # Expected errors are exactly the same as with TreeMesh!
+                        l2=nothing,   # [Float32(8.311947673061856e-6)],
+                        linf=nothing,)
+    # # Ensure that we do not have excessive memory allocations
+    # # (e.g., from type instabilities)
+    # let
+    #     t = sol.t[end]
+    #     u_ode = sol.u[end]
+    #     du_ode = similar(u_ode)
+    #     @test (@allocated Trixi.rhs!(du_ode, u_ode, semi, t)) < 1000
+    # end
+    @test real(ode.p.solver) == Float64
+    @test real(ode.p.solver.basis) == Float64
+    @test real(ode.p.solver.mortar) == Float64
+    # TODO: remake ignores the mesh itself as well
+    @test real(ode.p.mesh) == Float64
+
+    @test ode.u0 isa Array
+    @test ode.p.solver.basis.derivative_matrix isa Array
+
+    @test Trixi.storage_type(ode.p.cache.elements) === Array
+    @test Trixi.storage_type(ode.p.cache.interfaces) === Array
+    @test Trixi.storage_type(ode.p.cache.boundaries) === Array
+    @test Trixi.storage_type(ode.p.cache.mortars) === Array
+end
+
+@trixi_testset "elixir_advection_basic_gpu.jl Float32 / CUDA" begin
     # Using CUDA inside the testset since otherwise the bindings are hiddend by the anonymous modules
     using CUDA
-    # TODO(benegee/vchuravy): Port compute_coefficients! to KernelAbstractions.jl
-    CUDA.allowscalar(true)
     @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_basic_gpu.jl"),
                         # Expected errors are exactly the same as with TreeMesh!
                         l2=nothing,   # [Float32(8.311947673061856e-6)],
                         linf=nothing, # [Float32(6.627000273229378e-5)],
-                        RealT=Float32,
```

Comment:
We will need RealT once we have the test all running, but we can delete it for now.